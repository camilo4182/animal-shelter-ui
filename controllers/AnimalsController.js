import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: 'animals',
            data: animal
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(animal_name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${animal_name}`
        });
    }
}

