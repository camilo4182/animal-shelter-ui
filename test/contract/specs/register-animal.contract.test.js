import {provider} from "../specs/init-pact";
import {AnimalController} from "../../../controllers";
import {Matchers} from "@pact-foundation/pact";

describe('Given an animal service', () => {
    describe('When a request to register an animal is made', () => {
        const animal = {
            name: "Pelusa",
            breed: "Siames",
            gender: "Male",
            isVaccinated: true
        }
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'register an animal',
                uponReceiving: 'a request to register an animal',
                withRequest: {
                    method: 'POST',
                    path: '/animals',
                    headers:{
                        'Content-Type': 'application/json'
                    },
                    body: animal
                },
                willRespondWith: {
                    status: 201,
                    body: Matchers.like({
                        name: "Pelusa",
                        breed: "Siames",
                        gender: "Male",
                        isVaccinated: true
                    })
                }
            });
        });

        it('Then it should return the right http code', async () => {
            const response = await AnimalController.register(animal);
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        })
    });
});