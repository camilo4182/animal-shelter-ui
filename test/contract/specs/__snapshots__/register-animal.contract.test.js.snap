// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an animal service When a request to register an animal is made Then it should return the right http code 1`] = `
Array [
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "name": "manchas",
    "vaccinated": true,
  },
]
`;
